### DISCUSSION TITLE
Error: cannot import 'update' from 'pyqtgrapg.SRTTransform' - I checked it's there

### DISCUSSION THREAD

**Discussion by RT-Tap** [pyqtgraph/pyqtgraph#1879] [Q&A]
So I hate to be that guy to keep asking for help, but I tried searching and wasnt sure where else to look/ask.   

Anyway I was working on a little project of mine and it was going relatively smoothly, making changes, saving, running, debugging, saving and running again, just standard procedure, then all of a sudden I save and go to run the project literally 2 minutes after I just ran it and I get the following error: 

`Traceback (most recent call last):
  File "main.py", line 5, in <module>
    import OrderBookGraphs #type: ignore -this comment makes pylint ignore unresolved import error- file is importable
  File "C:\Users\Artyt\Documents\Python Scripts\gettingSomewhere\OrderBookGraphs.py", line 1, in <module>
    from pyqtgraph.SRTTransform import update
ImportError: cannot import name 'update' from 'pyqtgraph.SRTTransform' (C:\Users\Artyt\anaconda3\envs\gen\lib\site-packages\pyqtgraph\SRTTransform.py)`

I checked the file `SRTTransform.py` in  `C:\Users\Artyt\anaconda3\envs\gen\lib\site-packages\pyqtgraph\` and it seems to have an update function on line 161, so I went and tried to update/upgrade pyqtgraph with `conda install -c conda-forge  pyqtgraph` and also updated conda and still nothing.  

It's so weird everything was running fine, I made some simple changes and when I went to run it literally 2 or 3 minutes after I just ran it,  it just spits out this error.

Any and all help is greatly appreciated.

Can provide additional details as requested.

**Comment by RT-Tap** [comment]
Ok so the issue was I had a line at the very top that included `from pyqtgraph.SRTTransform import update` but I haven't changed the libraries to include since the project started so I'm not sure how that got there, but it was also 5am after a night of trying to debug different things so I may have missclicked something or I don't know but sorry for wasting your time.

You can delete this as it was a dumb mistake on my part.  

However now that I remove it, my code has some unexpected behavior so Ill try to figure that out on my own.

Actually, if anyone can let my know what SRTTransform does or more specifically what update from there does I'd appreciate it.

**Reply by j9ac9k** [reply]
`SRTTransform` is a subclass of QTransform which is a fairly low level class, and isn't used that much in this library (looks like it's used in ROI.py and CanvasItem (and CanvasItem doesn't have particularly widespread usage).  It provides some 

I'm not sure I understand your question, if you're not sure what SRTTransform does, why are you using it; or more specifically why are you using an update function from `SRTTransform.py`, which doesn't exist (there is an update method to SRTTransform class, but I'm not sure how that's relevant).

I would say if you're not doing some unusual transformations where you need to preserve a history or things of that nature, you probably shouldn't worry about that class.

**Reply by RT-Tap** [reply]
Honestly I wasnt trying to use it, it was 5am after a night of debugging and I must have clicked something or honestly I have no idea what I did but `from pyqtgraph.SRTTransform import update` just showed up as the first line in a class I wrote and that gave me problems.  I sure as shit didn't put that there and it's still a mystery to me as to how that got there.  I also noticed `from pyqtgraph.graphicsItems.PlotDataItem import dataType` just showed up as well so maybe I was just so tired I became delirious and don't' remember but everything is back to normal.  Maybe it was something I misclicked or entered in the IDE by accident, I just dont know.

Thanks for your response I really appreciate it.

**Reply by j9ac9k** [reply]
some editors with extensions have features that attempt to be helpful by automatically importing functions/modules when you use them in a file... that would be the one guess I have for something like that.  Anyway I'm going to mark this as closed.  Hope you track down the culprit, and try not to do too much coding at that kind of late hour, you spend twice as much time debugging your own work later 😆

**Comment by j9ac9k** [comment]
Going to make this one as closed due to a non-pyqtgraph issue, looks like some dev tool has gone awry and made changes in this code-base for some reason.
